library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt

```diff
@@ -81,91 +77,94 @@ internal class AuthAPIImpl : AuthAPI {
-
-        Amplify.Auth.signUp(
-            phone,
-            password,
-            options,
-            handler::onSuccess,
-            handler::onError,
-        )
+        return suspendCancellableCoroutine {
+            Amplify.Auth.signUp(
+                phone,
+                password,
+                options,
+                { resume(ApiResult.Success(Unit)) },
+                { resume(ApiResult.Failure(it)) }
+            )
+        }
     }
 
-    override fun signIn(
-        phoneNumber: String,
-        handler: AuthRequestHandler,
-    ) {
+    override suspend fun signIn(
+        phoneNumber: String
+    ): ApiResult<AuthSignInResult> {
         val authSignInOptions = AWSCognitoAuthSignInOptions.builder()
             .authFlowType(AuthFlowType.CUSTOM_AUTH)
             .build()
-
-        Amplify.Auth.signIn(
-            phoneNumber,
-            "",
-            authSignInOptions,
-            handler::onSuccess,
-            handler::onError,
-        )
+        return suspendCancellableCoroutine {
+            Amplify.Auth.signIn(
+                phoneNumber,
+                "",
+                authSignInOptions,
+                { resume(ApiResult.Success(it)) },
+                { resume(ApiResult.Failure(it)) }
+            )
+        }
     }
 
-    override fun confirmSignIn(
-        code: String,
-        handler: AuthRequestHandler
-    ) {
-        Amplify.Auth.confirmSignIn(
-            code,
-            handler::onSuccess,
-            handler::onError,
-        )
+    override suspend fun confirmSignIn(
+        code: String
+    ): ApiResult<AuthSignInResult> {
+        return suspendCancellableCoroutine {
+            Amplify.Auth.confirmSignIn(
+                code,
+                { resume(ApiResult.Success(it)) },
+                { resume(ApiResult.Failure(it)) }
+            )
+        }
     }
 
-    override fun confirmResendCode(
-        code: String,
-        handler: AuthRequestHandler
-    ) {
-        Amplify.Auth.confirmUserAttribute(
-            AuthUserAttributeKey.phoneNumber(),
-            code,
-            handler::onSuccess,
-            handler::onError
-        )
+    override suspend fun confirmResendCode(
+        code: String
+    ): ApiResult<Unit> {
+        return suspendCancellableCoroutine {
+            Amplify.Auth.confirmUserAttribute(
+                AuthUserAttributeKey.phoneNumber(),
+                code,
+                { resume(ApiResult.Success<Unit>(Unit)) },
+                { resume(ApiResult.Failure<Unit>(it)) },
+            )
+        }
     }
 
-    override fun sendCode(
+    override suspend fun sendCode(
         phoneNumber: String,
-        handler: AuthRequestHandler,
-    ) {
-        when (authenticationType) {
-            AuthenticationType.Mobile -> signIn(phoneNumber, handler)
-            is AuthenticationType.Facebook -> sendPhoneCodeByResend(handler)
+    ): ApiResult<Any> {
+        return when (authenticationType) {

```

Comment:
I believe, returning ApiResult<Any> is suboptimal, as it would later require us to cast the result to some type which we expect to receive. Currently we are using this function without expecting any parameters, so I suggest to change return type to ApiResult<Unit>

Reply:
done